libraries/chain/db_update.cpp

```diff
@@ -43,53 +43,24 @@
 
 namespace graphene { namespace chain {
 
-void database::update_global_dynamic_data( const signed_block& b )
+void database::update_global_dynamic_data( const signed_block& b, const uint32_t missed_blocks )
 {
    const dynamic_global_property_object& _dgp = dynamic_global_property_id_type(0)(*this);
    const global_property_object& gpo = get_global_properties();
 
-   uint32_t missed_blocks = get_slot_at_time( b.timestamp );
-
-//#define DIRTY_TRICK // problem with missed_blocks can occur when "maintenance_interval" set to few minutes
-#ifdef DIRTY_TRICK
-   if (missed_blocks != 0) {
-#else
-   assert( missed_blocks != 0 );
-#endif
-// bad if-condition, this code needs to execute for both shuffled and rng algorithms
-//     if (gpo.parameters.witness_schedule_algorithm == GRAPHENE_WITNESS_SHUFFLED_ALGORITHM)
-//     {
```

Comment:
@ronakpatel3110  : Can you explain why this code block is removed ?

Reply:
@bobinson The code to count the missed_blocks has been moved to new function `update_witness_missed_blocks`. Also, with the new approach to count missed blocks, there is no issue with maintenance interval as it is using re-shuffled list of witnesses. Re-shuffling happens after counting missed blocks, and does not happen while blocks are being missed.